### PR TITLE
First version of API without Unit tests

### PR DESCRIPTION
**Pull request by yoannpicquenot**

### CODE REVIEW

File: v1/index.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+});
+
+router.get('/gpus', (req, res) => {
+  gpu.find().then((doc) => {
+    res.send({
+      status: 200,
+      data: doc
+    });
+  }, (err) => {
+    console.log(err);
```
Comment (yoannpicquenot) [inline v1/index.js]: Not very useful

File: v1/index.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
+'use strict';
```
Comment (yoannpicquenot) [inline v1/index.js:1]: Four spaces of indentation

File: server.js [hunk 3]
```diff
@@ -0,0 +1,43 @@
+'use strict';
```
Comment (yoannpicquenot) [inline server.js:1]: Four spaces of indentation

File: package.json [hunk 4]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (mapy69003) [inline package.json]: Missing engines field.

File: index.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+var express = require('express');
```
Comment (mapy69003) [inline index.js]: Four spaces of indentation

File: config/index.js [hunk 6]
```diff
@@ -0,0 +1,6 @@
+module.exports = {
```
Comment (mapy69003) [inline config/index.js]: Four spaces of indentation

File: v1/models/gpu/index.js [hunk 7]
```diff
@@ -0,0 +1,226 @@
+var mongoose = require('mongoose');
```
Comment (yoannpicquenot) [inline v1/models/gpu/index.js:3]: Four spaces of indenation

**Review by yoannpicquenot — COMMENTED** [review]
- Console.log('') to remove
- Four spaces of indentation

**Comment by saveman71** [comment]
Waiting for #2

File: v1/index.js [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+    }, res.send);
+});
+
+router.get('/gpus/:id', (req, res) => {
+    let id = req.params.id;
+
+    gpu.findOne({
+        '_id': mongoose.Types.ObjectId(id)
+    }).then((doc) => {
+        res.send({
```
Comment (saveman71) [inline v1/index.js]: Why is this different than 10 lines above?

File: server.js [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+var mongoose = require('mongoose');
+var morgan = require('morgan');
+
+var config = require('./config');
+var server = express();
+
+var api = require('./index.js');
+
+mongoose.connect(config.mongodb_uri, (err) => {
+    console.log(err || ("Successfully connected to the db"));
```
Comment (saveman71) [inline server.js]: Un-necessary parens

File: v1/customTypes/formats.js [hunk 10]
```diff
@@ -0,0 +1,24 @@
+module.exports = function (mongoose) {
+    function Formats(key, options) {
+        mongoose.SchemaType.call(this, key, options, 'Formats');
+    };
+
+    Formats.prototype = Object.create(mongoose.SchemaType.prototype);
+
+    Formats.prototype.cast = function (val) {
+        let _val = val;
```
Comment (saveman71) [inline v1/customTypes/formats.js]: Why?

File: v1/customTypes/uint10.js [hunk 11]
```diff
@@ -0,0 +1,23 @@
+module.exports = function (mongoose) {
+    function Uint10(key, options) {
+        mongoose.SchemaType.call(this, key, options, 'Uint10');
+    };
+
+    Uint10.prototype = Object.create(mongoose.SchemaType.prototype);
+
+    Uint10.prototype.cast = function (val) {
+        let _val = Number(val);
```
Comment (saveman71) [inline v1/customTypes/uint10.js]: Why?

File: v1/customTypes/formats.js [hunk 12]
```diff
@@ -0,0 +1,24 @@
+module.exports = function (mongoose) {
```
Comment (saveman71) [inline v1/customTypes/formats.js]: Space after `function`

File: v1/customTypes/formats.js [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
+    function Formats(key, options) {
+        mongoose.SchemaType.call(this, key, options, 'Formats');
+    };
+
+    Formats.prototype = Object.create(mongoose.SchemaType.prototype);
+
+    Formats.prototype.cast = function (val) {
```
Comment (saveman71) [inline v1/customTypes/formats.js]: Space after `function`

File: v1/customTypes/uint10.js [hunk 14]
```diff
@@ -0,0 +1,23 @@
+module.exports = function (mongoose) {
```
Comment (saveman71) [inline v1/customTypes/uint10.js]: Space after `function`

File: v1/customTypes/uint10.js [hunk 15]
```diff
@@ -0,0 +1,23 @@
...
+    function Uint10(key, options) {
+        mongoose.SchemaType.call(this, key, options, 'Uint10');
+    };
+
+    Uint10.prototype = Object.create(mongoose.SchemaType.prototype);
+
+    Uint10.prototype.cast = function (val) {
```
Comment (saveman71) [inline v1/customTypes/uint10.js]: Space after `function`

File: v1/customTypes/uint12.js [hunk 16]
```diff
@@ -0,0 +1,23 @@
+module.exports = function (mongoose) {
```
Comment (saveman71) [inline v1/customTypes/uint12.js]: Space after `function`

File: v1/customTypes/uint12.js [hunk 17]
```diff
@@ -0,0 +1,23 @@
...
+    function Uint12(key, options) {
+        mongoose.SchemaType.call(this, key, options, 'Uint12');
+    };
+
+    Uint12.prototype = Object.create(mongoose.SchemaType.prototype);
+
+    Uint12.prototype.cast = function (val) {
```
Comment (saveman71) [inline v1/customTypes/uint12.js]: Space after `function`

File: server.js [hunk 18]
```diff
@@ -0,0 +1,43 @@
...
+
+    // CORS
+    server.use(function(req, res, next) {
```
Comment (saveman71) [inline server.js]: Arrow function

File: index.js [hunk 19]
```diff
@@ -0,0 +1,15 @@
...
+var router = express.Router();
+
+module.exports = function(req, res, next) {
+    try {
+        let version = require(`./${req.params.version}`);
+        version(req, res, next);
+    } catch (e) {
+        if (e.code == "MODULE_NOT_FOUND") {
+            res.sendFile(__dirname + '/notfound.html');
```
Comment (saveman71) [inline index.js]: Status is not 404

File: package.json [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+  "engines": {
+    "node": ">=6.10x"
+  },
+  "main": "server.js",
+  "scripts": {
+    "start": "node server.js",
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "Lugdunum3D",
+  "license": "ISC",
```
Comment (saveman71) [inline package.json]: MIT

Comment (saveman71) [reply package.json]: + add https://github.com/Lugdunum3D/glTF2-loader/blob/master/LICENSE at the root, replacing glTF2-loader with LugBench-API

File: server.js [hunk 21]
```diff
@@ -0,0 +1,43 @@
...
+
+var express = require('express');
+var bodyparser = require('body-parser');
+var mongoose = require('mongoose');
+var morgan = require('morgan');
+
+var config = require('./config');
+var server = express();
+
+var api = require('./index.js');
```
Comment (saveman71) [inline server.js]: `./index`

File: v1/customTypes/uint10.js (hunk 14 above)
Comment (saveman71) [inline v1/customTypes/uint10.js]: Add `'use strict';`

File: v1/customTypes/uint12.js (hunk 16 above)
Comment (saveman71) [inline v1/customTypes/uint12.js]: Add `'use strict';`

File: v1/models/gpu/index.js (hunk 7 above)
Comment (saveman71) [inline v1/models/gpu/index.js:3]: Add `'use strict';`

File: v1/customTypes/formats.js (hunk 12 above)
Comment (saveman71) [inline v1/customTypes/formats.js]: Add `'use strict';`

File: v1/index.js [hunk 22]
```diff
@@ -0,0 +1,43 @@
+'use strict';
+
+var express = require('express');
+var router = express.Router();
+var mongoose = require('mongoose');
```
Comment (saveman71) [inline v1/index.js]: Move mongoose before router

File: v1/models/gpu/index.js [hunk 23]
```diff
@@ -0,0 +1,226 @@
...
+require('../../customTypes/uint12.js')(mongoose);
+require('../../customTypes/uint10.js')(mongoose);
+require('../../customTypes/formats.js')(mongoose);
+
+module.exports = function() {
```
Comment (saveman71) [inline v1/models/gpu/index.js]: `module.exports = mongoose.model('Gpu', gpuSchema);` instead.

File: v1/models/gpu/index.js [hunk 24]
```diff
@@ -0,0 +1,226 @@
...
+            "queueFlags": [String],
+            "timestampValidBits": Number
+        }],
+        "extensions": [{
+            "extensionName": String,
+            "specVersion": Number
+        }],
+        "formats": [mongoose.Schema.Types.Formats]
+    });
+    return mongoose.model('gpu', schema);
```
Comment (saveman71) [inline v1/models/gpu/index.js]: `gpu` => `Gpu`, as schema should begin with an uppercase letter, do not change the folder though

**Review by saveman71 — CHANGES_REQUESTED** [review]

File: v1/customTypes/uint10.js (hunk 11 above)
Comment (yoannpicquenot) [reply v1/customTypes/uint10.js]: http://mongoosejs.com/docs/customschematypes.html

File: v1/customTypes/formats.js (hunk 10 above)
Comment (yoannpicquenot) [reply v1/customTypes/formats.js]: http://mongoosejs.com/docs/customschematypes.html

File: notfound.html [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+                position: absolute;
+                top: 50%;
+                line-height: 3em;
+                margin-top: -1.5em;
+            }
+
+        </style>
+        <div class="container">
+            <div class="content">
+                <h1>This api route doesn't exist</h1></div>
```
Comment (cchardeau) [inline notfound.html:37]: A 404 page with more style?

Comment (yoannpicquenot) [reply notfound.html:37]: Not really important. I made an update which removes this page.

**Pull request closed without merging**